crates/sui-types/src/effects/effects_v1.rs

```diff
@@ -176,6 +177,10 @@ impl TransactionEffectsAPI for TransactionEffectsV1 {
         &self.gas_used
     }
 
+    fn lamport_version(&self) -> SequenceNumber {
+        self.gas_object().0 .1
```

Comment:
Good callout. I'll update so we pass the computed lamport version through directly from the inner temporary store since that should be much less fragile.

Reply:
Sorry I know you have addressed it, but just in case if you want to go back and get lamport version from the effects, I think it's actually doable because I have merged effects v2 change. As long as shared object deletion is only enabled when effects v2 is enabled, lamport version will in fact be reliable